docker/gaffer/Dockerfile

```diff
@@ -31,23 +32,26 @@ WORKDIR /jars
 COPY ./files/ .
 # Try to download required version from Maven Central, otherwise build from source
 RUN	allFilesDownloaded="TRUE" && \
-	if [ ! -f "./accumulo-store-${GAFFER_VERSION}-iterators.jar" ]; then \
+    if echo "$BASE_IMAGE_TAG" | grep -q "^1.*$"; then allFilesDownloaded="FALSE"; fi && \
+	if [ ! -f "./accumulo-store-${GAFFER_VERSION}-iterators.jar" ] && [ "${allFilesDownloaded}" = "TRUE" ]; then \
 		wget -nv "${GAFFER_DOWNLOAD_URL}/uk/gov/gchq/gaffer/accumulo-store/${GAFFER_VERSION}/accumulo-store-${GAFFER_VERSION}-iterators.jar" || allFilesDownloaded="FALSE"; \
 	fi && \
 	for lib in $(echo ${GAFFER_LIBS} | tr , ' '); do \
-		if [ ! -f "./${lib}-${GAFFER_VERSION}-deploy.jar" ]; then \
+		if [ ! -f "./${lib}-${GAFFER_VERSION}-deploy.jar" ] && [ "${allFilesDownloaded}" = "TRUE" ]; then \
 			wget -nv "${GAFFER_DOWNLOAD_URL}/uk/gov/gchq/gaffer/${lib}/${GAFFER_VERSION}/${lib}-${GAFFER_VERSION}-deploy.jar" || allFilesDownloaded="FALSE"; \
 		fi \
 	done && \
 	if [ "${allFilesDownloaded}" = "FALSE" ]; then \
+        # Set legacy option based on BASE_IMAGE_TAG (Accumulo version)
+        if echo "$BASE_IMAGE_TAG" | grep -q "^2.*$"; then LEGACY=false; else LEGACY=true; fi && \
 		git clone ${GAFFER_GIT_REPO} /tmp/gaffer && \
 		cd /tmp/gaffer && \
-		git checkout ${GAFFER_VERSION} && \
+		git checkout ${GAFFER_VERSION} || git checkout gaffer2-${GAFFER_VERSION} && \
 		for dir in library/*; do if [ -f "${dir}/pom.xml" ]; then echo "${dir}"; fi done | \
 		paste -sd, | \
-		xargs mvn clean package -q -Pquick --also-make -pl store-implementation/accumulo-store/ -pl && \
+		xargs mvn clean package -Dlegacy=$LEGACY -q -Pquick --also-make -pl store-implementation/accumulo-store/ -pl && \
 		if [ ! -f "/jars/accumulo-store-${GAFFER_VERSION}-iterators.jar" ]; then \
-			cp ./store-implementation/accumulo-store/target/accumulo-store-*-iterators.jar /jars; \
+			cp ./store-implementation/accumulo-store/target/accumulo-store-*-iterators*.jar /jars; \
```

Comment:
Is the additional wildcard here significant / required? Earlier in the script when we try to obtain the jar from a Maven repo we only download a single iterators.jar

Reply:
This is a good point. There's a new jar ending in `iterators-legacy.jar`, this is built locally (with `-Dlegacy` flag) only and not downloaded. That is the significance to the new wildcard.